core/trino-server-rpm/src/main/resources/dist/etc/init.d/trino

```diff
@@ -37,34 +37,50 @@ CONFIGURATION=(
     --server-log-file "${SERVER_LOG_FILE:-/var/log/trino/server.log}"
 )
 
+declare -A CONFIG_ENV
 # shellcheck disable=1091
 source /etc/trino/env.sh
 
+join_by() {
+    local IFS="$1"
+    shift
+    echo "$*"
+}
+
+sudo_args=(-u "$SERVICE_USER")
+for key in "${!CONFIG_ENV[@]}"; do
+    export "$key=${CONFIG_ENV[$key]}"
+done
+if [ "${#CONFIG_ENV[@]}" -ne 0 ]; then
+    sudo_args+=(--preserve-env="$(join_by , "${!CONFIG_ENV[@]}")")
```

Comment:
The `env`-based approach can yield to simpler code, without `join_by` 
Did you try that? 

eg you could append env to `sudo_args` variable
(maybe rename to sudo_invocation?)

Reply:
Refactored to use `env` and renamed `args` to options, since that's how the manual of `env` refers to `key=value` pairs and it's not an abbreviation.